genie-agent/src/main/java/com/netflix/genie/agent/execution/services/UploadService.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services;
+
+import com.netflix.genie.agent.execution.exceptions.UploadException;
+
+import java.io.File;
+import java.net.URI;
+
+/**
+ * Service to upload local files to a target location.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+public interface UploadService {
+
+    /**
+     * Compress the file and upload to a target location.
+     *
+     * @param file                 file to upload
+     * @param targetLocationURI    URI representing the target location for the file
+     * @throws UploadException     if the upload fails
+     */
+    void compressAndUpload(final File file, final URI targetLocationURI) throws UploadException;
```

Comment:
API dissonates with name. Why not make 1 service to do 1 thing (upload in this case)?

Reply:
Can break it into `CompressService` and `UploadService` with only `upload()` method